kedro/config/logging.yml

```diff
@@ -32,16 +32,13 @@ handlers:
         delay: True
 
 loggers:
-    anyconfig:
-        level: WARNING
-        handlers: [console]
-        propagate: no
-
     kedro.framework.cli.hooks.manager:
         level: INFO
         handlers: [console]
         propagate: no
 
+    kedro:
+        level: INFO
+
```

Comment:
Originally in https://github.com/kedro-org/kedro/issues/1470 we needed to add `handlers` and `propagate` to all loggers. Why is this no longer needed? And what exactly do these things do? I still find the logging behaviour a bit mysterious 😅

Reply:
Yeah, I only fully understood this after reading the logging docs for the 1000th time last week 😀 This will become clearer with examples when we complete #1474 and I'm happy to present to the team to explain further, but in brief:

* loggers work hierarchically and activate all loggers above them. e.g. when `kedro.x.y` does logging it will also log to any loggers named `kedro.x` and `kedro`. `root` is the at the top of the hierarchy
* the above is the case when `propagate: true` (the default if `propagate` is not specified explicitly). If you set `propagate: false` then a logging call to `kedro.x.y` will not activate any other loggers
* if you propagate to higher up loggers then if they both have the same handler, this handler will be called separately by each logger that uses it. This results in duplicated log messages
* our general strategy according to the Python docs should be to delegate as much as possible to the highest up logger (so `root` if possible), attach handlers to that logger and then allow lower level handlers just to propagate their calls up the chain

The [docs on `propagate`](https://docs.python.org/3/library/logging.html#logging.Logger.propagate) are quite clear and helpful for explaining all this more.

In practice, what happens now is that any logging done anywhere inside kedro will be propagated up to the root logger, which attaches the required handlers. i.e. if `kedro.x.y` then it will check if there's any logger defined explicitly for `kedro.x.y` (there isn't), then it will check for `kedro.x` (there isn't), then `kedro` (which is defined). `kedro` doesn't define any handlers, so it gets propagates up to `root`, which is the thing that actually calls the handlers. The point in defining `kedro` as a logger is just so we can set `level: INFO` so that any info log messages coming from kedro are shown. The default level set by `root` (and hence by all other logging calls outside `kedro`) is WARNING.

`kedro.framework.cli.hooks.manager` is a weird special case that will hopefully disappear in the near future if we remove the file handlers from the root logger.

Reply:
Thanks for this explanation @AntonyMilneQB! It makes a lot more sense now 😄